smac/smac_cli.py

```diff
@@ -83,12 +85,43 @@ def main_cli(self):
                     fn=traj_fn, cs=scen.cs)
                 initial_configs.append(trajectory[-1]["incumbent"])
 
+        # Restore state
+        stats = None
+        incumbent = None
+        if args_.restore_state:
+            root_logger.debug("Restoring state from %s...", args_.restore_state)
+            # Check for folder and files
+            rh_path = os.path.join(args_.restore_state, "runhistory.json")
+            stats_path = os.path.join(args_.restore_state, "stats.json")
+            traj_path = os.path.join(args_.restore_state, "traj_aclib2.json")
+            scen_path = os.path.join(args_.restore_state, "scenario.txt")
+            if not os.path.isdir(args_.restore_state):
+               raise FileNotFoundError("Could not find folder from which to restore.")
+            # Load runhistory and stats
+            rh = RunHistory(aggregate_func=None)
+            rh.load_json(rh_path, scen.cs)
+            root_logger.debug("Restored runhistory from %s", rh_path)
+            stats = Stats(scen)
+            stats.load(stats_path)
+            root_logger.debug("Restored stats from %s", stats_path)
+            trajectory = TrajLogger.read_traj_aclib_format(
+                fn=traj_path, cs=scen.cs)
+            incumbent = trajectory[-1]["incumbent"]
+            root_logger.debug("Restored incumbent %s from %s", incumbent, traj_path)
+            # Copy traj if output_dir is different
+            if scen.output_dir != Scenario(scen_path).output_dir:
+                new_traj_path = os.path.join(scen.output_dir, "traj_aclib2.json")
+                shutil.copy(traj_path, new_traj_path)
```

Comment:
See command above... I figured that if the user specifies a new output_dir, then the restoring gets done in another folder. That way, the old version will be completely preserved.